manifests/init.pp

```diff
@@ -2,111 +2,112 @@
-    false, 'false': { $distroverpkg_bool = false } # lint:ignore:quoted_booleans
-    default:        { fail("yum::distroverpkg is not a boolean. It is <${distroverpkg}>.") }
-  }
-
-  validate_string(
-    $config_owner,
-    $config_group,
-    $repos_d_owner,
-    $repos_d_group,
-  )
-
-  if $proxy != undef {
-    validate_string($proxy)
-  }
-
-  if $pkgpolicy != undef {
-    validate_re("${$pkgpolicy}", '^(newest|last)$', # lint:ignore:only_variable_string
-      "yum::pkgpolicy must be <newest> or <last>. It is <${pkgpolicy}>.")
-  }
-
-  validate_re("${config_mode}", '^[0-7]{4}$', # lint:ignore:only_variable_string
-    "yum::config_mode is not a file mode in octal notation. It is <${config_mode}>.")
-
-  validate_re("${repos_d_mode}", '^[0-7]{4}$', # lint:ignore:only_variable_string
-    "yum::repos_d_mode is not a file mode in octal notation. It is <${config_mode}>.")
-
-
-  if $installonly_limit != undef {
-    case type3x($installonly_limit) {
-      'integer': { $installonly_limit_int = $installonly_limit }
-      'string':  { $installonly_limit_int = floor($installonly_limit) }
-      default:   { fail("yum::installonly_limit is not an integer nor stringified integer. It is <${installonly_limit}>.")}
-    }
-  }
-
-  $exclude_hiera_merge_bool = str2bool($exclude_hiera_merge)
-  validate_bool($exclude_hiera_merge_bool)
-
   if $exclude != undef {
-    case type3x($exclude) {
-      'array':  { $exclude_array = $exclude }
-      'string': { $exclude_array = any2array($exclude) }
-      default:  { fail("yum::exclude is <${exclude}> and must be either a string or an array.") }
+    case $exclude {
+      String:  { $exclude_array = any2array($exclude) }
+      default: { $exclude_array = $exclude }
     }
-
-    case $exclude_hiera_merge_bool {
+    case $exclude_hiera_merge {
       true:    { $exclude_array_real = hiera_array('yum::exclude') }
       default: { $exclude_array_real = $exclude_array }
     }
   }
 
-  case $manage_repos {
-    true, 'true': { # lint:ignore:quoted_booleans
-      $repos_d_purge   = true
-      $repos_d_recurse = true
-    }
-    false, 'false': { # lint:ignore:quoted_booleans
-      $repos_d_purge   = false
-      $repos_d_recurse = false
-    }
-    default: {
-      fail("yum::manage_repos is not a boolean. It is <${manage_repos}>.")
-    }
-  }
-
-  case $repos_hiera_merge {
-    true, 'true':   { $repos_hiera_merge_bool = true }  # lint:ignore:quoted_booleans
-    false, 'false': { $repos_hiera_merge_bool = false } # lint:ignore:quoted_booleans
-    default:        { fail("yum::repos_hiera_merge is not a boolean. It is <${repos_hiera_merge}>.") }
-  }
-
   if $repos != undef {
-    case $repos_hiera_merge_bool {
+    case $repos_hiera_merge {
       true:    { $repos_real = hiera_hash('yum::repos') }
       default: { $repos_real = $repos }
     }
     validate_hash($repos_real)
     create_resources('yum::repo',$repos_real)
```

Comment:
we should stop using create_resources and do this the Puppet4 way with lambda's.

https://github.com/sensu/sensu-puppet/blob/master/manifests/init.pp#L521-L525

Reply:
Can work on this in the next days.

Reply:
changed